clients/feeder/gps2val/gpsd_feeder.py

```diff
@@ -36,52 +36,20 @@ def __init__(self, config):
             print("kuksa_val section missing from configuration, exiting")
             sys.exit(-1)
         provider_config=config['kuksa_val']
-        self.sendMsgQueue = queue.Queue()
-        self.recvMsgQueue = queue.Queue()
-        self.client = VSSClientComm(self.sendMsgQueue, self.recvMsgQueue, provider_config)
+        self.client = VSSClientComm(provider_config)
         self.client.start()
         self.token = provider_config.get('token', "token.json")
-        self.authorize()
+        self.client.authorize(self.token)
         
-    def authorize(self):
-        if os.path.isfile(self.token):
-            with open(self.token, "r") as f:
-                self.token = f.readline()
-
-        req = {}
-        req["requestId"] = 1238
-        req["action"]= "authorize"
-        req["tokens"] = self.token
-        jsonDump = json.dumps(req)
-        self.sendMsgQueue.put(jsonDump)
-        #print(req)
-        resp = self.recvMsgQueue.get(timeout = 1)
-        #print(resp)
-
     def shutdown(self):
         self.client.stopComm()
 
-    def setValue(self, path, value):
-        if 'nan' == value:
-            print(path + " has an invalid value " + str(value))
-            return
-        req = {}
-        req["requestId"] = 1235
-        req["action"]= "set"
-        req["path"] = path
-        req["value"] = value
-        jsonDump = json.dumps(req)
-        #print(jsonDump)
-        self.sendMsgQueue.put(jsonDump)
-        resp = self.recvMsgQueue.get(timeout = 1)
-        #print(resp)
-        
     def setPosition(self, position):
-        self.setValue('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Altitude', position['alt'])
-        self.setValue('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Latitude', position["lat"])
-        self.setValue('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Longitude', position["lon"])
-        self.setValue('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Accuracy', position["hdop"])
-        self.setValue('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Speed', position["speed"])
+        self.client.setValue('Vehicle.Cabin.Infotainment.Navigation.CurrentLocation.Altitude', position['alt'])
```

Comment:
This will need to be changed soon, as the current positon is deprecated and will move to "Vehicle.Location"

Reply:
but not for now right?

Reply:
for now it works. For VSS 2.0 also. But for releases after 2.1 we need to change that. Just to keep in mind